### PR TITLE
Internal setting to change NP environments

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205236994263036/f

### Description
Dev settingn to change NP environment.

### Steps to test this PR
See [asana](https://app.asana.com/0/488551667048375/1205236994263036/f) task

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3434** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3434" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3434?utm_source=stack-comment).

File: network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/network/NetPInternalEnvDataStore.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
+    var netpCustomEnvironmentUrl: String?
+        get() = preferences.getString(KEY_NETP_ENVIRONMENT_URL, null)
+        set(value) = preferences.edit { putString(KEY_NETP_ENVIRONMENT_URL, value) }
+    var useNetpCustomEnvironmentUrl: Boolean
+        get() = preferences.getBoolean(KEY_NETP_USE_ENVIRONMENT_URL, false)
+        set(enabled) = preferences.edit { putBoolean(KEY_NETP_USE_ENVIRONMENT_URL, enabled) }
+
+    companion object {
+        private const val FILENAME = "com.duckduckgo.netp.internal.env.store.v1"
```
Comment (karlenDimla) [inline network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/network/NetPInternalEnvDataStore.kt:37]: Do we really need a separate file for this? I see that we have a separate shared prefs for MTU (`NetPInternalMtuProvider`) and system apps exclusion (`NetPInternalExclusionListProvider`)

**Review by karlenDimla — APPROVED** [review]
Tested and works as expected. 
Just one question around shared prefs files.

File: network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/network/NetPInternalEnvDataStore.kt (hunk 1 above)
Comment (aitorvs) [reply network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/network/NetPInternalEnvDataStore.kt:37]: ouch, I missed this question I was too quick merging. Sorry about that.

Yeah, probably we didn't need a separate shared prefs. Will take the opportunity of a next PR (whatever that may be) and fix this.

Sorry again!

**Pull request merged**